modules/internal/DefaultRectangular.chpl

```diff
@@ -114,10 +114,6 @@ module DefaultRectangular {
     proc dsiGetIndices() return ranges;
   
     proc dsiSetIndices(x) {
```

Comment:
I moved the checks into domain assignment following Brad’s suggestion in PR#3703.  He preferred having them hoisted into the domain to avoid making each distribution do the checks for themselves.  In my testing I didn’t find any places where the errors in domain assignment didn’t catch an error that dsiSetIndices used to issue.  That doesn’t prove that they can’t exist, but we don’t see any currently.